pkg/binding/controller.go

```diff
@@ -443,29 +424,29 @@ func (c *Controller) processNextWorkItem(ctx context.Context) bool {
 		// put back on the workqueue and attempted again after a back-off
 		// period.
 		defer c.workqueue.Done(obj)
-		var key util.Key
+		var objIdentifier util.ObjectIdentifier
 		var ok bool
-		// We expect util.Key to come off the workqueue. We do this as the delayed
-		// nature of the workqueue means the items in the informer cache may actually be
-		// more up to date that when the item was initially put onto the
-		// workqueue.
-		if key, ok = obj.(util.Key); !ok {
+		// We expect util.ObjectIdentifier to come off the workqueue. We do this as the
+		// delayed nature of the workqueue means the items in the informer
+		// cache may actually be more up to date that when the item was
+		// initially put onto the workqueue.
+		if objIdentifier, ok = obj.(util.ObjectIdentifier); !ok {
 			// if the item in the workqueue is invalid, we call
 			// Forget here to avoid process a work item that is invalid.
 			c.workqueue.Forget(obj)
-			utilruntime.HandleError(fmt.Errorf("expected util.Key in workqueue but got %#v", obj))
+			utilruntime.HandleError(fmt.Errorf("expected util.ObjectIdentifier in workqueue but got %#v", obj))
 			return nil
 		}
 		// Run the reconciler, passing it the full key or the metav1 Object
-		if err := c.reconcile(ctx, key); err != nil {
+		if err := c.reconcile(ctx, objIdentifier); err != nil {
 			// Put the item back on the workqueue to handle any transient errors.
 			c.workqueue.AddRateLimited(obj)
-			return fmt.Errorf("error syncing key '%#v': %s, requeuing", obj, err.Error())
+			return fmt.Errorf("error syncing object (identifier: %#v): %s, requeuing", objIdentifier, err.Error())
 		}
 		// Finally, if no error occurs we Forget this item so it does not
 		// get queued again until another change happens.
 		c.workqueue.Forget(obj)
-		logger.V(2).Info("Successfully synced", "object", obj)
+		logger.V(2).Info("Successfully synced", "object identifier", objIdentifier)
```

Comment:
I have seen loggers object to keys with spaces in them.

Reply:
Dang, I knew I should have been more specific. I wasn't, because I do not remember exactly _what_ was complaining. But the lesson I learned is that an Info key should look like a private identifier in Go: use only alphanumerics and maybe underscore, but lowerCamelCase is preferred.

Reply:
Updated.